### PR TITLE
8338014: Improve usage of @jvms tags in class file API

### PR DESCRIPTION
**Pull request by SoniaZaldana**
Hi all, 

This PR addresses [8338014](https://bugs.openjdk.org/browse/JDK-8338014) improving the use of `@jvms` tags by adding `JVMS` prior to the tag. 

Thanks, 
Sonia

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 2 [Reviewers](https://openjdk.org/bylaws#reviewer))

### Issue
 * [JDK-8338014](https://bugs.openjdk.org/browse/JDK-8338014): Improve usage of @jvms tags in class file API (**Bug** - P4)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20513/head:pull/20513` \
`$ git checkout pull/20513`

Update a local copy of the PR: \
`$ git checkout pull/20513` \
`$ git pull https://git.openjdk.org/jdk.git pull/20513/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20513`

View PR using the GUI difftool: \
`$ git pr show -t 20513`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20513.diff">https://git.openjdk.org/jdk/pull/20513.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20513#issuecomment-2284670253)

### CODE REVIEW

File: src/java.base/share/classes/java/lang/classfile/package-info.java
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/package-info.java:1]: These 2 changes are redundant and wrong. Please review your changes before submitting it after mechanical replacements.

**Review by jddarcy — APPROVED** [review]
Looks fine; increasing review count to also include someone who more directly maintains this API.

If you haven't done so already, I recommend also doing a quick check for an analagous issue with `@jls` tags in this API.

File: src/java.base/share/classes/java/lang/classfile/attribute/SyntheticAttribute.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 import jdk.internal.javac.PreviewFeature;
 
 /**
- * Models the {@code Synthetic} attribute {@jvms 4.7.8}, which can appear on
+ * Models the {@code Synthetic} attribute JVMS {@jvms 4.7.8}, which can appear on
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/attribute/SyntheticAttribute.java:37]: These occurrences should be `(JVMS {@jvms x.x.x})`.

File: src/java.base/share/classes/java/lang/classfile/constantpool/ConstantPoolBuilder.java [hunk 2]
```diff
@@ -365,7 +365,7 @@ default MethodHandleEntry methodHandleEntry(DirectMethodHandleDesc descriptor) {
      * it is returned; otherwise, a new entry is added and the new entry is
      * returned.
      *
-     * @param refKind the reference kind of the method handle {@jvms 4.4.8}
+     * @param refKind the reference kind of the method handle JVMS {@jvms 4.4.8}
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/constantpool/ConstantPoolBuilder.java:368]: Same, these occurrences are not part of a sentence, so enclose in parentheses like `(JVMS {@jvms 4.4.8})`

File: src/java.base/share/classes/java/lang/classfile/TypeAnnotation.java [hunk 3]
```diff
@@ -60,7 +60,7 @@
 import jdk.internal.javac.PreviewFeature;
 
 /**
- * Models an annotation on a type use, as defined in {@jvms 4.7.19} and {@jvms 4.7.20}.
+ * Models an annotation on a type use, as defined in JVMS {@jvms 4.7.19} and JVMS {@jvms 4.7.20}.
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/TypeAnnotation.java:63]:
```suggestion
 * Models an annotation on a type use, as defined in JVMS {@jvms 4.7.19} and {@jvms 4.7.20}.
```

File: src/java.base/share/classes/java/lang/classfile/Attribute.java [hunk 4]
```diff
@@ -66,7 +66,7 @@
 import jdk.internal.javac.PreviewFeature;
 
 /**
- * Models a classfile attribute {@jvms 4.7}.  Many, though not all, subtypes of
+ * Models a classfile attribute JVMS {@jvms 4.7}.  Many, though not all, subtypes of
```
Comment (liach) [inline src/java.base/share/classes/java/lang/classfile/Attribute.java:69]:
```suggestion
 * Models a classfile attribute (JVMS {@jvms 4.7}).  Many, though not all, subtypes of
```
Same as for all other attributes.

**Review by liach — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/lang/classfile/attribute/SyntheticAttribute.java (hunk 1 above)
Comment (jddarcy) [reply src/java.base/share/classes/java/lang/classfile/attribute/SyntheticAttribute.java:37]: I agree that structure would be better and more consistent with `@jmvs` usage elsewhere in core libs.

**Comment by jddarcy** [comment]
> @jddarcy Perhaps we should revisit the inline forms of `@jls` and `@jvms` tags to come up with similar and more consistent usage.

Certainly a variant do generate the most common usage pattern would be helpful -- brainstorming --

`{@jvms cite 1.2.3}`

to generate

(JVMVS 1.2.3)

where "1.2.3" was the expected link.

**Comment by liach** [comment]
I think we can just make `{@jvms }` inline versions emit JVMS and JLS prefix if there's some other text in the body, like `{@jvms 4.7.20 The {@code RuntimeVisibleTypeAnnotations} attribute}` may render

JVMS 4.7.20 The `RuntimeVisibleTypeAnnotations` attribute

**Comment by SoniaZaldana** [comment]
Hi @jddarcy, just checking in if the update looks good prior to integration.

**Review by jddarcy — APPROVED** [review]

**Comment by liach** [comment]
Please allow for about 24 hours after the last push so that everyone interested may review, such as @asotona.

**Review by asotona — APPROVED** [review]

**Pull request closed without merging**